### PR TITLE
fix!: update TPG 3.49, modules, and GKE for CI

### LINKED ISSUE #118: Deprecated K8's API's & Unsupported GKE Version.
### TL;DR

GKE Version in "defaults" (1.20) is no longer supported on GKE. Needs updating to minimum 1.23. 
Several K8'S API's have been deprecated an updated to full release in Kubernetes 1.23 and 1.24

### Expected behavior

Successful Deployment? 

### Observed behavior

Failed Deployment!

### Terraform Configuration

```hcl
Used Default Configuration provided within the Repo.
```

### Terraform Version

```sh
Terraform v1.3.0
on linux_amd64
```

### Additional information

None

### PR DESCRIPTION
**Pull request by apeabody**
Fix #118

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_documentation
 The documentation needs to be regenerated. Please run `make generate_docs`. 
 ```bash 
Checking for documentation generation
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' '--exclude=autogen' '--exclude=*.tfvars' /workspace/README.md /tmp/tmp.zVbhWhVNy2/generate_docs/workspace/README.md
53c53
< | gke\_version | Version of GKE to use for the GitLab cluster | `string` | `"1.20"` | no |
---
> | gke\_version | Version of GKE to use for the GitLab cluster | `string` | `"latest"` | no |
Error: Documentation generation has not been run, please run the
'make docker_generate_docs' command and commit the above changes.
 ```

**Comment by comment-bot-dev** [comment]
@apeabody
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_tflint
 Failed TFlint check. More details below. 
 ```bash 
Checking for tflint
Working in . ...
20 issue(s) found:
Warning: List items should be accessed using square brackets (terraform_deprecated_index)
  on main.tf line 365:
 365:   gitlab_address = var.gitlab_address_name == "" ? google_compute_address.gitlab.0.address : data.google_compute_address.gitlab.0.address
Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.1/docs/rules/terraform_deprecated_index.md
Warning: List items should be accessed using square brackets (terraform_deprecated_index)
  on main.tf line 365:
 365:   gitlab_address = var.gitlab_address_name == "" ? google_compute_address.gitlab.0.address : data.google_compute_address.gitlab.0.address
Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.1/docs/rules/terraform_deprecated_index.md
Warning: List items should be accessed using square brackets (terraform_deprecated_index)
  on main.tf line 365:
 365:   gitlab_address = var.gitlab_address_name == "" ? google_compute_address.gitlab.0.address : data.google_compute_address.gitlab.0.address
Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.1/docs/rules/terraform_deprecated_index.md
Warning: List items should be accessed using square brackets (terraform_deprecated_index)
  on main.tf line 365:
 365:   gitlab_address = var.gitlab_address_name == "" ? google_compute_address.gitlab.0.address : data.google_compute_address.gitlab.0.address
Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.1/docs/rules/terraform_deprecated_index.md
Warning: Missing version constraint for provider "time" in "required_providers" (terraform_required_providers)
  on main.tf line 383:
 383: resource "time_sleep" "sleep_for_cluster_fix_helm_6361" {
Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.1/docs/rules/terraform_required_providers.md
Warning: `project_id` variable has no type (terraform_typed_variables)
  on variables.tf line 17:
  17: variable "project_id" {
Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.1/docs/rules/terraform_typed_variables.md
Warning: `domain` variable has no type (terraform_typed_variables)
  on variables.tf line 21:
  21: variable "domain" {
Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.1/docs/rules/terraform_typed_variables.md
Warning: `certmanager_email` variable has no type (terraform_typed_variables)
  on variables.tf line 26:
  26: variable "certmanager_email" {
Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.1/docs/rules/terraform_typed_variables.md
Warning: `gke_version` variable has no type (terraform_typed_variables)
  on variables.tf line 30:
  30: variable "gke_version" {
Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.1/docs/rules/terraform_typed_variables.md
Warning: `gke_machine_type` variable has no type (terraform_typed_variables)
  on variables.tf line 35:
  35: variable "gke_machine_type" {
Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.1/docs/rules/terraform_typed_variables.md
Warning: `gitlab_db_name` variable has no type (terraform_typed_variables)
  on variables.tf line 40:
  40: variable "gitlab_db_name" {
Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.1/docs/rules/terraform_typed_variables.md
Warning: `gitlab_db_random_prefix` variable has no type (terraform_typed_variables)
  on variables.tf line 45:
  45: variable "gitlab_db_random_prefix" {
Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.1/docs/rules/terraform_typed_variables.md
Warning: `gitlab_deletion_protection` variable has no type (terraform_typed_variables)
  on variables.tf line 50:
  50: variable "gitlab_deletion_protection" {
Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.1/docs/rules/terraform_typed_variables.md
Warning: `gitlab_db_password` variable has no type (terraform_typed_variables)
  on variables.tf line 55:
  55: variable "gitlab_db_password" {
Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.1/docs/rules/terraform_typed_variables.md
Warning: `gitlab_address_name` variable has no type (terraform_typed_variables)
  on variables.tf line 60:
  60: variable "gitlab_address_name" {
Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.1/docs/rules/terraform_typed_variables.md
Warning: `gitlab_runner_install` variable has no type (terraform_typed_variables)
  on variables.tf line 65:
  65: variable "gitlab_runner_install" {
Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.1/docs/rules/terraform_typed_variables.md
Warning: `region` variable has no type (terraform_typed_variables)
  on variables.tf line 70:
  70: variable "region" {
Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.1/docs/rules/terraform_typed_variables.md
Warning: `gitlab_nodes_subnet_cidr` variable has no type (terraform_typed_variables)
  on variables.tf line 75:
  75: variable "gitlab_nodes_subnet_cidr" {
Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.1/docs/rules/terraform_typed_variables.md
Warning: `gitlab_pods_subnet_cidr` variable has no type (terraform_typed_variables)
  on variables.tf line 80:
  80: variable "gitlab_pods_subnet_cidr" {
Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.1/docs/rules/terraform_typed_variables.md
Warning: `gitlab_services_subnet_cidr` variable has no type (terraform_typed_variables)
  on variables.tf line 85:
  85: variable "gitlab_services_subnet_cidr" {
Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.1/docs/rules/terraform_typed_variables.md
tflint failed . 
Working in ./examples/simple_example ...
tflint passed ./examples/simple_example 
Skipping ./test/fixtures/simple_example
Skipping ./test/setup
 ```

**Comment by comment-bot-dev** [comment]
@apeabody
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_documentation
 The documentation needs to be regenerated. Please run `make generate_docs`. 
 ```bash 
Checking for documentation generation
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' '--exclude=autogen' '--exclude=*.tfvars' /workspace/README.md /tmp/tmp.el9vHIdkUH/generate_docs/workspace/README.md
55c55
< | project\_id | GCP Project to deploy resources | `any` | n/a | yes |
---
> | project\_id | GCP Project to deploy resources | `string` | n/a | yes |
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' '--exclude=autogen' '--exclude=*.tfvars' /workspace/examples/simple_example/README.md /tmp/tmp.el9vHIdkUH/generate_docs/workspace/examples/simple_example/README.md
10c10
< | project\_id | The project ID to deploy to | `any` | n/a | yes |
---
> | project\_id | The project ID to deploy to | `string` | n/a | yes |
Error: Documentation generation has not been run, please run the
'make docker_generate_docs' command and commit the above changes.
 ``` 
- ⚠️check_tflint
 Failed TFlint check. More details below. 
 ```bash 
Checking for tflint
Working in . ...
1 issue(s) found:
Warning: `certmanager_email` variable has no type (terraform_typed_variables)
  on variables.tf line 28:
  28: variable "certmanager_email" {
Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.1/docs/rules/terraform_typed_variables.md
tflint failed . 
Working in ./examples/simple_example ...
tflint passed ./examples/simple_example 
Skipping ./test/fixtures/simple_example
Skipping ./test/setup
 ```

**Comment by comment-bot-dev** [comment]
@apeabody
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_documentation
 The documentation needs to be regenerated. Please run `make generate_docs`. 
 ```bash 
Checking for documentation generation
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' '--exclude=autogen' '--exclude=*.tfvars' /workspace/README.md /tmp/tmp.oNX9bElBTM/generate_docs/workspace/README.md
54c54
< | helm\_chart\_version | Helm chart version to install during deployment | `string` | `"4.2.4"` | no |
---
> | helm\_chart\_version | Helm chart version to install during deployment | `string` | `"6.6.2"` | no |
Error: Documentation generation has not been run, please run the
'make docker_generate_docs' command and commit the above changes.
 ```

**Comment by comment-bot-dev** [comment]
@apeabody
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@apeabody
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@apeabody
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@apeabody
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@apeabody
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@apeabody
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Review by bharathkkb — APPROVED** [review]
Thanks @apeabody

**Comment by comment-bot-dev** [comment]
@apeabody
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: variables.tf [hunk 1]
```diff
@@ -16,74 +16,89 @@
...
 variable "certmanager_email" {
   description = "Email used to retrieve SSL certificates from Let's Encrypt"
+  type        = string
 }
 
 variable "gke_version" {
   description = "Version of GKE to use for the GitLab cluster"
-  default     = "1.20"
+  type        = string
+  default     = "1.21"
```
[the remark this answers is not in the document]
Comment (apeabody) [reply variables.tf:36]: @bharathkkb GKE 1.21 is the most recent that works with the current Gitlab Helm chart version, so a larger update will be needed to address.

**Pull request merged**